README.md

```diff
@@ -1 +1,30 @@
-# indexer
\ No newline at end of file
+# indexer
+
+## Configuration
+You can configure the application in 3 ways.
+The order of priority is
+1. Command line arguments
+2. Environment variables
+3. Configuration files
+
+### Configuration using configuration files
+The default configuration file is `configs/config.yml`. You can use the `--config` flag to specify a different configuration file.  
+If you want to add secrets to the configuration file, you can copy `configs/secrets.example.yml` to `configs/secrets.yml` and add the secrets.
```

Comment:
can we make the secrets files accept a `--secrets` for the path of the secrets file

Reply:
Do you need it? Currently thinking about it like this: If you want to add secrets, just put them in the `configs/secrets` file and it won't be commited and works.
If you want to pass in a separate config file, you can add regular config + any secrets you have there

Otherwise, which takes precedence? If you only declare a secrets file, does it override whatever is in the default config file and provided config file? I think it adds much more complexity.